packages/discord.js/src/structures/Message.js

```diff
@@ -30,340 +30,216 @@ const Util = require('../util/Util');
  * @extends {Base}
  */
 class Message extends Base {
-  constructor(client, data) {
+  constructor(client, data = {}) {
     super(client);
-
-    /**
-     * The id of the channel the message was sent in
-     * @type {Snowflake}
-     */
-    this.channelId = data.channel_id;
-
-    /**
-     * The id of the guild the message was sent in, if any
-     * @type {?Snowflake}
-     */
-    this.guildId = data.guild_id ?? this.channel?.guild?.id ?? null;
-
+    this.data.channel_id = data.channel_id;
+    this.data.guild_id = data.guild_id ?? this.channel?.guild?.id ?? null;
     this._patch(data);
```

Comment:
`this.data` is undefined here. 
```suggestion
    this._patch(data);
    this.data.channel_id = data.channel_id;
    this.data.guild_id = data.guild_id ?? this.channel?.guild?.id ?? null;
```

will fix the issue ig